### ISSUE TITLE
FTBFS in SPIRVDuplicatesTracker.cpp of release/15.x

### ISSUE THREAD

**Issue by kpajko79** [llvm/llvm-project#57686]
Hello,

While trying to compile the experimental SPIRV backend, an error is reported:

```console
ld.lld: error: undefined symbol: llvm::MachineInstr::dump() const
```

The method is unconditionally present in `MachineInstr.h`, but not in `MachineInstr.cpp`:

```cpp
#if !defined(NDEBUG) || defined(LLVM_ENABLE_DUMP)
```

The issue is already fixed on the main branch in 59351fe340f20a605bae53260ed30a8e0fd95cb6.
Could you please cherry-pick it to the release branch?
